packages/dev/core/src/Engines/engine.ts

```diff
@@ -1275,7 +1279,7 @@ export class Engine extends ThinEngine {
             }
         }
 
-        if (this._activeRenderLoops.length > 0) {
+        if (this._activeRenderLoops.length > 0 && this._frameHandler === -1) {

```

Comment:
In what case is not -1?

Reply:
if during the frame you call cancel then runRenderLoop again with a different function for instance. This is indeed the main issue discovered on the associated forum thread.